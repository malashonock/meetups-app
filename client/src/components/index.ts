export * from './Typography/Typography';
export * from './Button/Button';
export * from './MeetupCard/MeetupCard';
export * from './UserPreview/UserPreview';
export * from './VotesCount/VotesCount';
export * from './IconButton/IconButton';
export * from './DeleteButton/DeleteButton';
export * from './EditButton/EditButton';
export * from './Header/Header';
export * from './Tooltip/Tooltip';
export * from './InputField/InputField';
export * from './TextInput/TextInput';
export * from './TextField/TextField';
export * from './TextArea/TextArea';
export * from './HelperText/HelperText';
export * from './InputLabel/InputLabel';
export * from './Tabs/Tabs';
export * from './NavTabs/NavTabs';
export * from './Tab/Tab';
export * from './TabContent/TabContent';
export * from './TabsManager/TabsManager';
export * from './TabsIndicator/TabsIndicator';
export * from './MeetupStagesTabs/MeetupStagesTabs';
export * from './MeetupStagesTabs/MeetupTabContent/MeetupTabContent';
export * from './MeetupStagesTabs/meetupTabsMapper';
export * from './NewsCard/NewsCard';
export * from './DateTimePicker/DateTimePicker';
export * from './Stepper/Stepper';
export * from './Step/Step';
export * from './StepperProgress/StepperProgress';
export * from './ImageDropbox/ImageDropbox';
export * from './ImagePreview/ImagePreview';
export * from './ImageUploader/ImageUploader';
export * from './AuthToggle/AuthToggle';
export * from './SelectField/SelectField';
export * from './RootStoreProvider/RootStoreProvider';
export * from './LanguageSelect/LanguageSelect';
export * from './ProtectedRoute/ProtectedRoute';
export * from './AvatarGroup/AvatarGroup';
export * from './LoadingSpinner/LoadingSpinner';
export * from './Portal/Portal';
export * from './Modal/Modal';
export * from './ConfirmDialog/ConfirmDialog';
export * from './ConfirmDialogProvider/ConfirmDialogProvider';
export * from './Toast/Toast';
export * from './AlertStack/AlertStack';
