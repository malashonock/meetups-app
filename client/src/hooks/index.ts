export * from './useMeetupQuery';
export * from './useNewsArticleQuery';
export * from './useStaticFileQuery';
export * from './useAuthStore';
