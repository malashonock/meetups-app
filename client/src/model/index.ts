export * from './meetup.model';
export * from './news.model';
export * from './user.model';
export * from './login.model';
