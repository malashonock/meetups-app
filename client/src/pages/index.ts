export * from './LoginPage/LoginPage';
export * from './MeetupsPage/MeetupsPage';
export * from './ViewMeetupPage/ViewMeetupPage';
export * from './CreateMeetupPage/CreateMeetupPage';
export * from './EditMeetupPage/EditMeetupPage';
export * from './NewsPage/NewsPage';
export * from './ViewNewsPage/ViewNewsPage';
export * from './CreateNewsPage/CreateNewsPage';
export * from './EditNewsPage/EditNewsPage';
export * from './NotFoundPage/NotFoundPage';
