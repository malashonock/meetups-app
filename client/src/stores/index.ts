export * from './root.store';
export * from './auth.store';
export * from './user.store';
export * from './ui.store';
export * from './meetup.store';
export * from './news.store';
