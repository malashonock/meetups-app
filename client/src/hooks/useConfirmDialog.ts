import { useContext } from 'react';

import { ConfirmDialogContext, ConfirmFn } from 'components';
import { Optional } from 'types';

export const useConfirmDialog = (): ConfirmFn => {
  return useContext(ConfirmDialogContext);
};
