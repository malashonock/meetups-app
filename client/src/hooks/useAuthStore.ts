import { useContext } from 'react';

import { AuthStore } from 'stores';
import { RootContext } from 'components';

export const useAuthStore = (): AuthStore => {
  return useContext(RootContext).authStore;
};
