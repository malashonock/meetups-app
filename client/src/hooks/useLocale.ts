import { useContext } from 'react';

import { UiStore, Locale } from 'stores';
import { RootContext } from 'components';
import { Optional } from 'types';

export type UseLocaleResult = [
  locale: Optional<Locale>,
  setLocale: (locale: Locale) => void,
];

export const useLocale = (): UseLocaleResult => {
  const uiStore: Optional<UiStore> = useContext(RootContext)?.uiStore;

  const locale = uiStore?.locale;

  const setLocale = (locale: Locale): void => {
    if (uiStore) {
      uiStore.locale = locale;
    }
  };

  return [locale, setLocale];
};
