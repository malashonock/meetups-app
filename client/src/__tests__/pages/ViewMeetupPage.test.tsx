import { PropsWithChildren } from 'react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ViewMeetupPage } from 'pages';
import {
  mockTopic,
  mockMeetupDraft,
  mockMeetupDraftFilled,
  mockMeetup,
  mockTopicData,
  mockMeetupData,
  mockFullUser,
  mockFullUser2,
  mockUser,
  mockUser2,
} from 'model/__fakes__';
import { useAuthStore, useMeetup } from 'hooks';
import { Meetup, RootStore } from 'stores';
import { ConfirmDialogProvider } from 'components';
import { IMeetup } from 'model';

// Mock useAuthStore & useMeetup hook
jest.mock('hooks', () => {
  return {
    ...jest.requireActual('hooks'),
    useAuthStore: jest.fn(),
    useMeetup: jest.fn(),
  };
});
const mockUseAuthStore = useAuthStore as jest.MockedFunction<
  typeof useAuthStore
>;
const mockUseMeetup = useMeetup as jest.MockedFunction<typeof useMeetup>;

const mockMeetupApprove = jest.spyOn(Meetup.prototype, 'approve');
const mockMeetupPublish = jest.spyOn(Meetup.prototype, 'publish');
const mockMeetupDelete = jest.spyOn(Meetup.prototype, 'delete');
const mockMeetupVote = jest.spyOn(Meetup.prototype, 'vote');
const mockMeetupWithDrawVote = jest.spyOn(Meetup.prototype, 'withdrawVote');
const mockMeetupJoin = jest.spyOn(Meetup.prototype, 'join');
const mockMeetupCancelJoin = jest.spyOn(Meetup.prototype, 'cancelJoin');

afterEach(() => {
  jest.resetAllMocks();
});

const MockRouter = ({ children }: PropsWithChildren): JSX.Element => (
  <ConfirmDialogProvider>
    <MemoryRouter initialEntries={['/meetups', '/meetups/aaa']}>
      <Routes>
        <Route path="/meetups">
          <Route index element={<h1>Meetups page</h1>} />
          <Route path=":id">
            <Route index element={children} />
            <Route path="edit" element={<h1>Edit meetup</h1>} />
          </Route>
        </Route>
      </Routes>
    </MemoryRouter>
  </ConfirmDialogProvider>
);

describe('ViewMeetupPage', () => {
  describe('before topic approval', () => {
    beforeEach(() => {
      mockUseMeetup.mockReturnValue({
        meetup: mockTopic,
        isLoading: false,
        isError: false,
        errors: [],
      });
    });

    describe('given no user is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: null,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('given a user with non-admin rights is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: mockFullUser2,
        });
      });

      describe('Support Topic button', () => {
        describe('given the logged user has not yet voted', () => {
          beforeEach(() => {
            const rootStore = new RootStore();
            rootStore.authStore.loggedUser = mockFullUser2;
            const topic = new Meetup(mockTopicData, rootStore.meetupStore);
            mockUseMeetup.mockReturnValue({
              meetup: topic,
              isLoading: false,
              isError: false,
              errors: [],
            });
          });

          it('should match snapshot', () => {
            const { asFragment } = render(<ViewMeetupPage />, {
              wrapper: MockRouter,
            });
            expect(asFragment()).toMatchSnapshot();
          });

          it('on click, should call meetup.vote() method', async () => {
            render(<ViewMeetupPage />, { wrapper: MockRouter });

            userEvent.click(screen.getByText('viewMeetupPage.supportTopicBtn'));

            await waitFor(() => {
              expect(mockMeetupVote).toHaveBeenCalled();
            });
          });
        });

        describe('given the logged user already voted', () => {
          beforeEach(() => {
            const rootStore = new RootStore();
            rootStore.authStore.loggedUser = mockFullUser2;
            const topicData: IMeetup = {
              ...mockTopicData,
              votedUsers: [mockUser, mockUser2],
            };
            const topic = new Meetup(topicData, rootStore.meetupStore);
            mockUseMeetup.mockReturnValue({
              meetup: topic,
              isLoading: false,
              isError: false,
              errors: [],
            });
          });

          it('should match snapshot', () => {
            const { asFragment } = render(<ViewMeetupPage />, {
              wrapper: MockRouter,
            });
            expect(asFragment()).toMatchSnapshot();
          });

          it('on click, should call meetup.withdrawVote() method', async () => {
            render(<ViewMeetupPage />, { wrapper: MockRouter });

            userEvent.click(
              screen.getByText('viewMeetupPage.unsupportTopicBtn'),
            );

            await waitFor(() => {
              expect(mockMeetupWithDrawVote).toHaveBeenCalled();
            });
          });
        });
      });
    });

    describe('given a user with admin rights is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: mockFullUser,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });

      describe('Delete button', () => {
        it('on click, should call meetup.delete() method', async () => {
          render(<ViewMeetupPage />, { wrapper: MockRouter });

          userEvent.click(screen.getByText('formButtons.delete'));
          userEvent.click(screen.getByTestId('confirm-button'));

          await waitFor(() => {
            expect(mockMeetupDelete).toHaveBeenCalled();
          });
        });
      });

      describe('Approve Topic button', () => {
        it('on click, should call meetup.approve() method', () => {
          render(<ViewMeetupPage />, { wrapper: MockRouter });
          userEvent.click(screen.getByText('viewMeetupPage.approveTopicBtn'));
          expect(mockMeetupApprove).toHaveBeenCalled();
        });
      });
    });
  });

  describe('after topic approval, but before publishing', () => {
    beforeEach(() => {
      mockUseMeetup.mockReturnValue({
        meetup: mockMeetupDraftFilled,
        isLoading: false,
        isError: false,
        errors: [],
      });
    });

    describe('given no user is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: null,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('given a user with non-admin rights is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: mockFullUser2,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('given a user with admin rights is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: mockFullUser,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });

      describe('Publish button', () => {
        describe('given time & location have been filled out', () => {
          it('on click, should call meetup.publish() method', () => {
            render(<ViewMeetupPage />, { wrapper: MockRouter });
            userEvent.click(screen.getByText('formButtons.publish'));
            expect(mockMeetupPublish).toHaveBeenCalled();
          });
        });

        describe('given time or location have not been filled out', () => {
          beforeEach(() => {
            mockUseMeetup.mockReturnValue({
              meetup: mockMeetupDraft,
              isLoading: false,
              isError: false,
              errors: [],
            });
          });

          it('should be disabled', () => {
            render(<ViewMeetupPage />, { wrapper: MockRouter });
            expect(screen.getByText('formButtons.publish')).toBeDisabled();
          });

          it('invalid time/location message should be displayed', () => {
            render(<ViewMeetupPage />, { wrapper: MockRouter });
            expect(
              screen.getByText('viewMeetupPage.timePlaceInfo.invalid'),
            ).toBeInTheDocument();
          });
        });
      });
    });
  });

  describe('after approval & publishing', () => {
    beforeEach(() => {
      mockUseMeetup.mockReturnValue({
        meetup: mockMeetup,
        isLoading: false,
        isError: false,
        errors: [],
      });
    });

    describe('given no user is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: null,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('given a user with non-admin rights is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: mockFullUser2,
        });
      });

      describe('given meetup start has not passed', () => {
        describe('Join Meetup toggler', () => {
          describe('given the logged user has not yet joined', () => {
            beforeEach(() => {
              const rootStore = new RootStore();
              rootStore.authStore.loggedUser = mockFullUser2;
              const start: Date = new Date(2999, 0, 1);
              const meetupData: IMeetup = {
                ...mockMeetupData,
                start,
                finish: new Date(start.getTime() + 2 * 60 * 60 * 1000),
              };
              const meetup = new Meetup(meetupData, rootStore.meetupStore);
              mockUseMeetup.mockReturnValue({
                meetup,
                isLoading: false,
                isError: false,
                errors: [],
              });
            });

            it('should match snapshot', () => {
              const { asFragment } = render(<ViewMeetupPage />, {
                wrapper: MockRouter,
              });
              expect(asFragment()).toMatchSnapshot();
            });

            it('on click, should call meetup.join() method', async () => {
              render(<ViewMeetupPage />, { wrapper: MockRouter });

              userEvent.click(screen.getByText('viewMeetupPage.joinMeetupBtn'));

              await waitFor(() => {
                expect(mockMeetupJoin).toHaveBeenCalled();
              });
            });
          });

          describe('given the logged user already joined', () => {
            beforeEach(() => {
              const rootStore = new RootStore();
              rootStore.authStore.loggedUser = mockFullUser2;
              const start = new Date(2999, 0, 1);
              const meetupData: IMeetup = {
                ...mockMeetupData,
                start,
                finish: new Date(start.getTime() + 2 * 60 * 60 * 1000),
                participants: [mockUser, mockUser2],
              };
              const meetup = new Meetup(meetupData, rootStore.meetupStore);
              mockUseMeetup.mockReturnValue({
                meetup,
                isLoading: false,
                isError: false,
                errors: [],
              });
            });

            it('should match snapshot', () => {
              const { asFragment } = render(<ViewMeetupPage />, {
                wrapper: MockRouter,
              });
              expect(asFragment()).toMatchSnapshot();
            });

            it('on click, should call meetup.cancelJoin() method', async () => {
              render(<ViewMeetupPage />, { wrapper: MockRouter });

              userEvent.click(
                screen.getByText('viewMeetupPage.unjoinMeetupBtn'),
              );

              await waitFor(() => {
                expect(mockMeetupCancelJoin).toHaveBeenCalled();
              });
            });
          });
        });
      });

      describe('given meetup start has passed', () => {
        it('should match snapshot', () => {
          const { asFragment } = render(<ViewMeetupPage />, {
            wrapper: MockRouter,
          });
          expect(asFragment()).toMatchSnapshot();
        });
      });
    });

    describe('given a user with admin rights is logged in', () => {
      beforeEach(() => {
        mockUseAuthStore.mockReturnValue({
          loggedUser: mockFullUser,
        });
      });

      it('should match snapshot', () => {
        const { asFragment } = render(<ViewMeetupPage />, {
          wrapper: MockRouter,
        });
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('regardless of meetup status and user role', () => {
    beforeEach(() => {
      mockUseAuthStore.mockReturnValue({
        loggedUser: mockFullUser,
      });
      mockUseMeetup.mockReturnValue({
        meetup: mockMeetup,
        isLoading: false,
        isError: false,
        errors: [],
      });
    });

    describe('Back button', () => {
      it('should navigate to previous page', async () => {
        render(<ViewMeetupPage />, { wrapper: MockRouter });
        userEvent.click(screen.getByText('formButtons.back'));
        expect(screen.getByText('Meetups page')).toBeInTheDocument();
      });
    });

    it('should render a Loading spinner if meetup is undefined', () => {
      mockUseMeetup.mockReturnValue({});
      render(<ViewMeetupPage />, { wrapper: MockRouter });
      expect(screen.getByText('loadingText.meetup')).toBeInTheDocument();
    });

    it('should render a Loading spinner while meetup is loading', () => {
      mockUseMeetup.mockReturnValue({
        meetup: mockMeetup,
        isLoading: true,
      });
      render(<ViewMeetupPage />, { wrapper: MockRouter });
      expect(screen.getByText('loadingText.meetup')).toBeInTheDocument();
    });

    it('should render Not Found page if an error occurred while loading the meetup', () => {
      mockUseMeetup.mockReturnValue({
        meetup: mockMeetup,
        isError: true,
      });
      render(<ViewMeetupPage />, { wrapper: MockRouter });
      expect(screen.getByText('notFoundPage.title')).toBeInTheDocument();
    });
  });
});
