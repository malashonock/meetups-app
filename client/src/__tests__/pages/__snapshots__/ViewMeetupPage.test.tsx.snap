// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMeetupPage after approval & publishing given a user is logged in should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container CONFIRMED"
  >
    <h1
      class="typography h1 heading"
    >
      viewMeetupPage.title
    </h1>
    <div
      class="dataWrapper"
    >
      <div
        class="headerData"
      >
        <figure
          class="imageWrapper"
        >
          <img
            alt="viewMeetupPage.imgAlt"
            class="image"
            src="http://localhost/test.jpg"
          />
        </figure>
        <div
          class="headerDataContent"
        >
          <h2
            class="typography h2 meetupHeading"
          >
            Test meetup topic
          </h2>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.timePlaceInfo.valid
        </span>
        <div
          class="dataContent timePlaceInfo"
        >
          <div
            class="info"
            id="date"
          >
            <img
              alt="viewMeetupPage.date"
              class="image"
              src="calendar.svg"
            />
            <span
              class="typography span"
            >
              m, 15 марта, 2023
            </span>
          </div>
          <div
            class="info"
            id="time"
          >
            <img
              alt="viewMeetupPage.time"
              class="image"
              src="clock.svg"
            />
            <span
              class="typography span"
            >
              14:00 — 15:30
            </span>
          </div>
          <div
            class="info"
            id="location"
          >
            <img
              alt="viewMeetupPage.location"
              class="image"
              src="pin.svg"
            />
            <span
              class="typography span"
            >
              room 123
            </span>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.speakers
        </span>
        <div
          class="dataContent"
        >
          <div
            class="speakerWrapper"
          >
            <div
              class="user default"
              data-testid="user-preview"
            >
              <div
                class="avatar"
              >
                <span
                  class="typography span initials"
                >
                  JD
                </span>
              </div>
              <span
                class="typography span name noWrap"
              >
                John Doe
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.description
        </span>
        <div
          class="dataContent"
        >
          <p
            class="typography p excerpt"
          >
            Test meetup description
          </p>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.supporters
        </span>
        <div
          class="dataContent"
        >
          <ul
            class="container justified"
            style="--gap: 10px;"
          />
        </div>
      </div>
      <div
        class="dataContent actions"
      >
        <button
          class="actionButton backBtn button default"
          id="btn-back"
        >
          formButtons.back
        </button>
        <div
          class="actionsWrapper"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ViewMeetupPage after approval & publishing given no user is logged in should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container CONFIRMED"
  >
    <h1
      class="typography h1 heading"
    >
      viewMeetupPage.title
    </h1>
    <div
      class="dataWrapper"
    >
      <div
        class="headerData"
      >
        <figure
          class="imageWrapper"
        >
          <img
            alt="viewMeetupPage.imgAlt"
            class="image"
            src="http://localhost/test.jpg"
          />
        </figure>
        <div
          class="headerDataContent"
        >
          <h2
            class="typography h2 meetupHeading"
          >
            Test meetup topic
          </h2>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.timePlaceInfo.valid
        </span>
        <div
          class="dataContent timePlaceInfo"
        >
          <div
            class="info"
            id="date"
          >
            <img
              alt="viewMeetupPage.date"
              class="image"
              src="calendar.svg"
            />
            <span
              class="typography span"
            >
              m, 15 марта, 2023
            </span>
          </div>
          <div
            class="info"
            id="time"
          >
            <img
              alt="viewMeetupPage.time"
              class="image"
              src="clock.svg"
            />
            <span
              class="typography span"
            >
              14:00 — 15:30
            </span>
          </div>
          <div
            class="info"
            id="location"
          >
            <img
              alt="viewMeetupPage.location"
              class="image"
              src="pin.svg"
            />
            <span
              class="typography span"
            >
              room 123
            </span>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.speakers
        </span>
        <div
          class="dataContent"
        >
          <div
            class="speakerWrapper"
          >
            <div
              class="user default"
              data-testid="user-preview"
            >
              <div
                class="avatar"
              >
                <span
                  class="typography span initials"
                >
                  JD
                </span>
              </div>
              <span
                class="typography span name noWrap"
              >
                John Doe
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.description
        </span>
        <div
          class="dataContent"
        >
          <p
            class="typography p excerpt"
          >
            Test meetup description
          </p>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.supporters
        </span>
        <div
          class="dataContent"
        >
          <ul
            class="container justified"
            style="--gap: 10px;"
          />
        </div>
      </div>
      <div
        class="dataContent actions"
      >
        <button
          class="actionButton backBtn button default"
          id="btn-back"
        >
          formButtons.back
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ViewMeetupPage after topic approval, but before publishing given a user is logged in should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container DRAFT"
  >
    <h1
      class="typography h1 heading"
    >
      viewMeetupPage.title
    </h1>
    <div
      class="dataWrapper"
    >
      <div
        class="headerData"
      >
        <figure
          class="imageWrapper"
        >
          <img
            alt="viewMeetupPage.imgAlt"
            class="image"
            src="http://localhost/test.jpg"
          />
        </figure>
        <div
          class="headerDataContent"
        >
          <h2
            class="typography h2 meetupHeading"
          >
            Test meetup topic
          </h2>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.timePlaceInfo.valid
        </span>
        <div
          class="dataContent timePlaceInfo"
        >
          <div
            class="info"
            id="date"
          >
            <img
              alt="viewMeetupPage.date"
              class="image"
              src="calendar.svg"
            />
            <span
              class="typography span"
            >
              m, 15 марта, 2023
            </span>
          </div>
          <div
            class="info"
            id="time"
          >
            <img
              alt="viewMeetupPage.time"
              class="image"
              src="clock.svg"
            />
            <span
              class="typography span"
            >
              14:00 — 15:30
            </span>
          </div>
          <div
            class="info"
            id="location"
          >
            <img
              alt="viewMeetupPage.location"
              class="image"
              src="pin.svg"
            />
            <span
              class="typography span"
            >
              room 123
            </span>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.speakers
        </span>
        <div
          class="dataContent"
        >
          <div
            class="speakerWrapper"
          >
            <div
              class="user default"
              data-testid="user-preview"
            >
              <div
                class="avatar"
              >
                <span
                  class="typography span initials"
                >
                  JD
                </span>
              </div>
              <span
                class="typography span name noWrap"
              >
                John Doe
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.description
        </span>
        <div
          class="dataContent"
        >
          <p
            class="typography p excerpt"
          >
            Test meetup description
          </p>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.supporters
        </span>
        <div
          class="dataContent"
        >
          <ul
            class="container justified"
            style="--gap: 10px;"
          />
        </div>
      </div>
      <div
        class="dataContent actions"
      >
        <button
          class="actionButton backBtn button default"
          id="btn-back"
        >
          formButtons.back
        </button>
        <div
          class="actionsWrapper"
        >
          <button
            class="actionButton publishBtn button primary"
            id="btn-publish"
          >
            formButtons.publish
          </button>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ViewMeetupPage after topic approval, but before publishing given no user is logged in should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container DRAFT"
  >
    <h1
      class="typography h1 heading"
    >
      viewMeetupPage.title
    </h1>
    <div
      class="dataWrapper"
    >
      <div
        class="headerData"
      >
        <figure
          class="imageWrapper"
        >
          <img
            alt="viewMeetupPage.imgAlt"
            class="image"
            src="http://localhost/test.jpg"
          />
        </figure>
        <div
          class="headerDataContent"
        >
          <h2
            class="typography h2 meetupHeading"
          >
            Test meetup topic
          </h2>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.timePlaceInfo.valid
        </span>
        <div
          class="dataContent timePlaceInfo"
        >
          <div
            class="info"
            id="date"
          >
            <img
              alt="viewMeetupPage.date"
              class="image"
              src="calendar.svg"
            />
            <span
              class="typography span"
            >
              m, 15 марта, 2023
            </span>
          </div>
          <div
            class="info"
            id="time"
          >
            <img
              alt="viewMeetupPage.time"
              class="image"
              src="clock.svg"
            />
            <span
              class="typography span"
            >
              14:00 — 15:30
            </span>
          </div>
          <div
            class="info"
            id="location"
          >
            <img
              alt="viewMeetupPage.location"
              class="image"
              src="pin.svg"
            />
            <span
              class="typography span"
            >
              room 123
            </span>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.speakers
        </span>
        <div
          class="dataContent"
        >
          <div
            class="speakerWrapper"
          >
            <div
              class="user default"
              data-testid="user-preview"
            >
              <div
                class="avatar"
              >
                <span
                  class="typography span initials"
                >
                  JD
                </span>
              </div>
              <span
                class="typography span name noWrap"
              >
                John Doe
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.description
        </span>
        <div
          class="dataContent"
        >
          <p
            class="typography p excerpt"
          >
            Test meetup description
          </p>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.supporters
        </span>
        <div
          class="dataContent"
        >
          <ul
            class="container justified"
            style="--gap: 10px;"
          />
        </div>
      </div>
      <div
        class="dataContent actions"
      >
        <button
          class="actionButton backBtn button default"
          id="btn-back"
        >
          formButtons.back
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ViewMeetupPage before topic approval given a user is logged in should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container REQUEST"
  >
    <h1
      class="typography h1 heading"
    >
      viewMeetupPage.title
    </h1>
    <div
      class="dataWrapper"
    >
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.topic
        </span>
        <div
          class="dataContent"
        >
          <h2
            class="typography h2 meetupHeading"
          >
            Test meetup topic
          </h2>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.author
        </span>
        <div
          class="dataContent"
        >
          <div
            class="user default"
            data-testid="user-preview"
          >
            <div
              class="avatar"
            >
              <span
                class="typography span initials"
              >
                JD
              </span>
            </div>
            <span
              class="typography span name noWrap"
            >
              John Doe
            </span>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.description
        </span>
        <div
          class="dataContent"
        >
          <p
            class="typography p excerpt"
          >
            Test meetup description
          </p>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.supporters
        </span>
        <div
          class="dataContent"
        >
          <ul
            class="container justified"
            style="--gap: 10px;"
          />
        </div>
      </div>
      <div
        class="dataContent actions"
      >
        <button
          class="actionButton backBtn button default"
          id="btn-back"
        >
          formButtons.back
        </button>
        <div
          class="actionsWrapper"
        >
          <button
            class="actionButton deleteBtn button secondary"
            id="btn-delete"
          >
            formButtons.delete
          </button>
          <button
            class="actionButton approveBtn button primary"
            id="btn-approve"
          >
            viewMeetupPage.approveTopicBtn
          </button>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ViewMeetupPage before topic approval given no user is logged in should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container REQUEST"
  >
    <h1
      class="typography h1 heading"
    >
      viewMeetupPage.title
    </h1>
    <div
      class="dataWrapper"
    >
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.topic
        </span>
        <div
          class="dataContent"
        >
          <h2
            class="typography h2 meetupHeading"
          >
            Test meetup topic
          </h2>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.author
        </span>
        <div
          class="dataContent"
        >
          <div
            class="user default"
            data-testid="user-preview"
          >
            <div
              class="avatar"
            >
              <span
                class="typography span initials"
              >
                JD
              </span>
            </div>
            <span
              class="typography span name noWrap"
            >
              John Doe
            </span>
          </div>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.description
        </span>
        <div
          class="dataContent"
        >
          <p
            class="typography p excerpt"
          >
            Test meetup description
          </p>
        </div>
      </div>
      <div
        class="data"
      >
        <span
          class="typography span dataName"
        >
          viewMeetupPage.supporters
        </span>
        <div
          class="dataContent"
        >
          <ul
            class="container justified"
            style="--gap: 10px;"
          />
        </div>
      </div>
      <div
        class="dataContent actions"
      >
        <button
          class="actionButton backBtn button default"
          id="btn-back"
        >
          formButtons.back
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
